android/src/com/mapswithme/util/Utils.java

```diff
@@ -289,4 +289,10 @@ public static void navigateToParent(Activity activity)
     else
       NavUtils.navigateUpFromSameTask(activity);
   }
+
+  public static boolean directoryExists(String path)
+  {
+    final File file = new File(path);
+    return (file.exists() && file.isDirectory() && file.canRead());
```

Comment:
Are you sure that canRead() is ok if you want to use this code for our *writable* dir? And canWrite() in this case is not enough too due to a bug: https://code.google.com/p/android/issues/detail?id=66369
We had special check - tried to create/delete temporary file from the path to confirm that it can be used as an external storage.